src/apps/frontend/contexts/auth.provider.tsx

```diff
@@ -2,30 +2,47 @@ import React, {
   createContext,
   PropsWithChildren,
   useContext,
+  useState,
 } from 'react';
 
 import { AuthService } from '../services';
 import {
-  AccessToken, ApiResponse, AsyncError,
+  AccessToken, ApiResponse, AsyncError, PhoneNumber,
 } from '../types';
 
 import useAsync from './async.hook';
 
 type AuthContextType = {
   isLoginLoading: boolean;
+  isSendOtpLoading: boolean;
   isSignupLoading: boolean;
   isUserAuthenticated: () => boolean;
+  isVerifyOtpLoading: boolean;
   login: (username: string, password: string) => Promise<AccessToken>;
   loginError: AsyncError;
   loginResult: AccessToken;
   logout: () => void;
+  phoneNumber: PhoneNumber;
+  sendOtp: (
+    countryCode: string,
+    phoneNumber: string,
+  ) => Promise<void>;
+  sendOtpError: AsyncError;
+  setPhoneNumber: React.Dispatch<React.SetStateAction<PhoneNumber>>;
```

Comment:
No need to maintain phone number in state. When page moves from enter phone number to OTP page, pass this in URL to make the URL's self sufficient. Similarly, there is no need to expose `phoneNumber: PhoneNumber;`